supportedBackends/sfs/src/main/scala/cromwell/backend/sfs/SharedFileSystemAsyncJobExecutionActor.scala

```diff
@@ -266,6 +291,44 @@ trait SharedFileSystemAsyncJobExecutionActor
           |""".stripMargin)
   }
 
+  /**
+    * Send the job id of the running job to the key value store.
+    *
+    * @param runningJobOption The running job.
+    */
+  private def tellKvJobId(runningJobOption: SharedFileSystemJob): Unit = {
+    val kvJobKey =
+      KvJobKey(jobDescriptor.key.call.fullyQualifiedName, jobDescriptor.key.index, jobDescriptor.key.attempt)
+    val scopedKey = ScopedKey(jobDescriptor.workflowDescriptor.id, kvJobKey, SharedFileSystemJob.JobIdKey)
+    val kvValue = Option(runningJobOption.jobId)
+    val kvPair = KvPair(scopedKey, kvValue)
+    val kvPut = KvPut(kvPair)
+    serviceRegistryActor ! kvPut
+  }
+
+  def recoverScript(job: SharedFileSystemJob): ExecutionHandle = {
+    context.become(sharedReceive(Option(job)) orElse super.receive)
+    // To avoid race conditions, check for the rc file after checking if the job is alive.
+    if (isAlive(job) || jobPaths.returnCode.exists) {
+      // If we're done, we'll get to the rc during the next poll.
+      // Or if we're still running, return pending also.
+      jobLogger.info(s"Recovering using job id: ${job.jobId}")
+      SharedFileSystemPendingExecutionHandle(jobDescriptor, job)
+    } else {
+      // Could start executeScript(), but for now fail because we shouldn't be in this state.
+      FailedNonRetryableExecutionHandle(new RuntimeException(
+        s"Unable to determine that ${job.jobId} is alive, and ${jobPaths.returnCode.fullPath} does not exist."), None)
+    }
+  }
+
+  def isAlive(job: SharedFileSystemJob): Boolean = {
+    val argv = checkAliveArgs(job).argv
+    val stdout = pathPlusSuffix(jobPaths.stdout, "check")
+    val stderr = pathPlusSuffix(jobPaths.stderr, "check")
+    val checkAlive = new ProcessRunner(argv, stdout, stderr)
```

Comment:
ToL: (I'm only saying this because I used the pattern recently for file hashes and docker lookup)(more a "what do **you** think" comment than a suggestion or request)

How would you feel about repositioning all of these command line calls behind a global CommandLineRunnerActor router so there's an explicit rate limit?